### PR TITLE
(Android) Expose module to save symptom logs

### PR DESCRIPTION
**Pull request by juancruzgs**
#### Description:

Adds `SymptomLogEntryModule` and changes the field types of existing `CheckIns`.
It replicates the changes made on iOS in this PR https://github.com/Path-Check/gaen-mobile/pull/435

### CODE REVIEW

**Review by aledustet — APPROVED** [review]
🚢 :shipit: 

We are gonna need a delete functionality for the SYmptom Log entries, if you think is better to add it here go for it but it's ok if you want to merge this and add a new PR. Thanks!

**Comment by juancruzgs** [comment]
@aledustet I just added it here https://github.com/Path-Check/gaen-mobile/pull/439/commits/e3ef8007d21825736bddada6817bf1d1b9c29b20, let me know if that works for you

File: src/gaen/nativeModule.ts [hunk 1]
```diff
@@ -272,3 +272,7 @@ export const modifyLogEntry = (entry: SymptomLogEntry): Promise<void> => {
 export const getLogEntries = (): Promise<SymptomLogEntry[]> => {
   return symptomLogEntryModule.getSymptomLogEntries()
 }
+
+export const deleteLogEntry = (entry: SymptomLogEntry): Promise<void> => {
```
Comment (aledustet) [inline src/gaen/nativeModule.ts]: I think we made it receive only the id as a string 😬

**Review by aledustet — APPROVED** [review]
Looks good, but the signature of the native module and the methods is different, on one we are sending the object and on another one the id. We are using the id on the IOS implementation.

File: src/gaen/nativeModule.ts [hunk 2]
```diff
@@ -272,3 +272,7 @@ export const modifyLogEntry = (entry: SymptomLogEntry): Promise<void> => {
...
+  return symptomLogEntryModule.deleteSymptomLogEntry(entry.id)
```
Comment (juancruzgs) [inline src/gaen/nativeModule.ts]: @aledustet It sends the ID to the native layer, but I can also use the ID in the JS signature if that's easier for you

Comment (aledustet) [reply src/gaen/nativeModule.ts]: No problem with that! I believe we added the method with the right signature so you might be even able to remove it altogether 😄

**Pull request merged**